bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/handler/IpCameraHandler.java

```diff
@@ -0,0 +1,1999 @@
+                    if (ffmpegMjpeg != null) {
+                        ffmpegMjpeg.stopConverting();
+                    }
+                }
+            }
+        }
+    }
+
+    // sends direct to ctx so can be either snapshots.mjpeg or normal mjpeg stream
+    public void sendMjpegFirstPacket(ChannelHandlerContext ctx) {
+        final String BOUNDARY = "thisMjpegStream";
+        String contentType = "multipart/x-mixed-replace; boundary=" + BOUNDARY;
+        HttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+        response.headers().add(HttpHeaderNames.CONTENT_TYPE, contentType);
+        response.headers().set(HttpHeaderNames.CACHE_CONTROL, HttpHeaderValues.NO_CACHE);
+        response.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+        response.headers().add("Access-Control-Allow-Origin", "*");
+        response.headers().add("Access-Control-Expose-Headers", "*");
+        ctx.channel().writeAndFlush(response);
+    }
+
+    public void sendMjpegFrame(byte[] jpg, ChannelGroup channelGroup) {
+        final String BOUNDARY = "thisMjpegStream";
+        ByteBuf imageByteBuf = Unpooled.copiedBuffer(jpg);
+        int length = imageByteBuf.readableBytes();
+        String header = "--" + BOUNDARY + "\r\n" + "content-type: image/jpeg" + "\r\n" + "content-length: " + length
+                + "\r\n\r\n";
+        ByteBuf headerBbuf = Unpooled.copiedBuffer(header, 0, header.length(), StandardCharsets.UTF_8);
+        ByteBuf footerBbuf = Unpooled.copiedBuffer("\r\n", 0, 2, StandardCharsets.UTF_8);
+        streamToGroup(headerBbuf, channelGroup, false);
+        streamToGroup(imageByteBuf, channelGroup, false);
+        streamToGroup(footerBbuf, channelGroup, true);
+    }
+
+    public void streamToGroup(Object msg, ChannelGroup channelGroup, boolean flush) {
+        channelGroup.write(msg);
+        if (flush) {
+            channelGroup.flush();
+        }
+    }
+
+    private void storeSnapshots() {
+        int count = 0;
+        // Need to lock as fifoSnapshotBuffer is not thread safe and new snapshots can be incoming.
+        lockCurrentSnapshot.lock();
+        try {
+            for (Object incomingJpeg : fifoSnapshotBuffer) {
+                byte[] foo = (byte[]) incomingJpeg;
+                File file = new File(ffmpegOutputFolder + "snapshot" + count + ".jpg");
+                count++;
+                try {
+                    OutputStream fos = new FileOutputStream(file);
+                    fos.write(foo);
+                    fos.close();
+                } catch (FileNotFoundException e) {
+                    logger.warn("FileNotFoundException {}", e.getMessage());
+                } catch (IOException e) {
+                    logger.warn("IOException {}", e.getMessage());
+                }
+            }
+        } finally {
+            lockCurrentSnapshot.unlock();
+        }
+    }
+
+    public void setupFfmpegFormat(String format) {
```

Comment:
Since `format` should only be a small finite number of values, you should make this parameter a enum type instead.

Reply:
Done.